modules/@apostrophecms/doc-type/ui/apos/components/AposDocContextMenu.vue

```diff
@@ -0,0 +1,297 @@
+<template>
+  <AposContextMenu
+    class="apos-admin-bar_context-button"
+    :menu="menu"
+    :disabled="disabled || (menu.length === 0)"
+    menu-placement="bottom-end"
+    @item-clicked="menuHandler"
+    @open="$emit('menu-open')"
+    @close="$emit('menu-close')"
+    :button="{
+      tooltip: { content: 'More Options', placement: 'bottom' },
+      label: 'More Options',
+      icon: 'dots-vertical-icon',
+      iconOnly: true,
+      type: 'subtle',
+      modifiers: ['small', 'no-motion']
+    }"
+  />
+</template>
+
+<script>
+import { detectDocChange } from 'Modules/@apostrophecms/schema/lib/detectChange';
+import AposPublishMixin from 'Modules/@apostrophecms/ui/mixins/AposPublishMixin';
+import AposArchiveMixin from 'Modules/@apostrophecms/ui/mixins/AposArchiveMixin';
+import klona from 'klona';
+
+export default {
+  name: 'AposDocContextMenu',
+  mixins: [ AposPublishMixin, AposArchiveMixin ],
+  props: {
+    doc: {
+      type: Object,
+      required: true
+    },
+    // If editing in a modal, pass the current value object from the editor here
+    // so that the visibility of options takes unsaved changes into account
+    current: {
+      type: Object,
+      default() {
+        return null;
+      }
+    },
+    // Existing published version of doc, if there is one
+    published: {
+      type: Object,
+      default() {
+        return null;
+      }
+    },
+    // These props all default to true, and can be passed with a value of false to hide
+    // functionality even if the user can otherwise perform the action on this document.
+    // The component will figure out on its own if the user can perform each action or not.
+    showEdit: {
+      type: Boolean,
+      default() {
+        return true;
+      }
+    },
+    showPreview: {
+      type: Boolean,
+      default() {
+        return true;
+      }
+    },
+    showDiscardDraft: {
+      type: Boolean,
+      default() {
+        return true;
+      }
+    },
+    showCopy: {
+      type: Boolean,
+      default() {
+        return true;
+      }
+    },
+    showArchive: {
+      type: Boolean,
+      default() {
+        return true;
+      }
+    },
+    showRestore: {
+      type: Boolean,
+      default() {
+        return true;
+      }
+    },
+    showDismissSubmission: {
+      type: Boolean,
+      default() {
+        return true;
+      }
+    },
+    disabled: {
```

Comment:
The component will disable itself if it determines that no options are available right now. This prop is for going beyond that to say that the calling context wants it disabled no matter what.